pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,500 @@
+	}
+	if cfg.P2P.ProtoTickInterval == 0 {
+		cfg.P2P.ProtoTickInterval = 2 * time.Second
+	}
+	if cfg.TraceableChainLength == 0 {
+		cfg.TraceableChainLength = 2102400
+	}
+	if cfg.P2P.Ping.Interval == 0 {
+		cfg.P2P.Ping.Interval = 30 * time.Second
+	}
+	if cfg.P2P.Ping.Timeout == 0 {
+		cfg.P2P.Ping.Timeout = time.Minute
+	}
+
+	nodeAcc, err := utilConfig.LoadAccount(cfg.Wallet.Path, "", cfg.Wallet.Password)
+	if err != nil {
+		return nil, fmt.Errorf("read node account: %w", err)
+	}
+
+	standByCommittee := make([]string, len(cfg.Committee))
+	for i := range cfg.Committee {
+		standByCommittee[i] = hex.EncodeToString(cfg.Committee[i].Bytes())
+	}
+
+	var cfgBase config.Config
+
+	cfgBaseProto := &cfgBase.ProtocolConfiguration
+	cfgBaseProto.Magic = cfg.NetworkMagic
+	cfgBaseProto.StandbyCommittee = standByCommittee
+
+	cfgBaseProto.TimePerBlock = cfg.BlockInterval
+	cfgBaseProto.SeedList = cfg.SeedNodes
+	cfgBaseProto.VerifyTransactions = true
+	cfgBaseProto.P2PSigExtensions = !cfg.NotaryDisabled
+	cfgBaseProto.MaxTraceableBlocks = cfg.TraceableChainLength
+	cfgBaseProto.Hardforks = cfg.HardForks
+	if cfg.ValidatorsHistory != nil {
+		cfgBaseProto.ValidatorsHistory = make(map[uint32]int, len(cfg.ValidatorsHistory))
+		for height, num := range cfg.ValidatorsHistory {
+			cfgBaseProto.ValidatorsHistory[height] = int(num)
+		}
+	} else {
+		cfgBaseProto.ValidatorsCount = len(standByCommittee)
+	}
+	cfgBaseProto.NativeUpdateHistories = cfg.NativeActivations
+
+	cfgBaseApp := &cfgBase.ApplicationConfiguration
+	cfgBaseApp.Relay = true
+	cfgBaseApp.UnlockWallet = cfg.Wallet
+	cfgBaseApp.Consensus.Enabled = true
+	cfgBaseApp.Consensus.UnlockWallet = cfg.Wallet
+	cfgBaseApp.P2PNotary.Enabled = !cfg.NotaryDisabled
+	cfgBaseApp.P2PNotary.UnlockWallet = cfg.Wallet
+	cfgBaseApp.RPC.Enabled = true
+	cfgBaseApp.RPC.Addresses = cfg.RPCListenAddresses
+	cfgBaseApp.RPC.StartWhenSynchronized = true
+	cfgBaseApp.RPC.MaxGasInvoke = fixedn.Fixed8FromInt64(100)
+	cfgBaseApp.P2P.Addresses = cfg.P2P.ListenAddresses
+	cfgBaseApp.P2P.DialTimeout = cfg.P2P.DialTimeout
+	cfgBaseApp.P2P.ProtoTickInterval = cfg.P2P.ProtoTickInterval
+	cfgBaseApp.P2P.PingInterval = cfg.P2P.Ping.Interval
+	cfgBaseApp.P2P.PingTimeout = cfg.P2P.Ping.Timeout
+	cfgBaseApp.P2P.MinPeers = int(cfg.P2P.MinPeers)
+	cfgBaseApp.P2P.AttemptConnPeers = int(cfg.P2P.AttemptConnPeers)
+	cfgBaseApp.P2P.MaxPeers = int(cfg.P2P.MaxPeers)
+
+	err = cfgBase.ProtocolConfiguration.Validate()
+	if err != nil {
+		return nil, fmt.Errorf("invalid composed protocol configuration: %w", err)
+	}
+
+	cfgServer, err := network.NewServerConfig(cfgBase)
+	if err != nil {
+		return nil, fmt.Errorf("compose NeoGo server config from the base one: %w", err)
+	}
+
+	var cfgDB dbconfig.DBConfiguration
+	cfgDB.Type = cfg.Storage.typ
+	if cfgDB.Type == dbconfig.BoltDB {
+		cfgDB.BoltDBOptions.FilePath = cfg.Storage.path
+	} else if cfgDB.Type == dbconfig.LevelDB {
+		cfgDB.LevelDBOptions.DataDirectoryPath = cfg.Storage.path
+	}
```

Comment:
You don't need these if-else, just fill both level's and bolt's paths.

Reply:
Why? Then one of the statements will definitely be no-op, isn't it?